packages/noya-state/src/reducers/applicationReducer.ts

```diff
@@ -224,6 +226,32 @@ export function applicationReducer(
     case 'goToSymbolSource':
     case 'setOverrideValue':
       return symbolsReducer(state, action);
+    case 'setExportScale':
+    case 'setExportName':
+    case 'setExportFileFormat':
+    case 'setExportNamingScheme':
+    case 'addExportFormat':
+    case 'deleteExportFormat':
+      const pageIndex = getCurrentPageIndex(state);
+      const layerIndexPaths = getSelectedLayerIndexPaths(state);
+
+      return produce(state, (draft) => {
+        accessPageLayers(draft, pageIndex, layerIndexPaths).forEach((layer) => {
+          layer.exportOptions = exportReducer(layer.exportOptions, action);
```

Comment:
Maybe we could pass the `layer.frame` into `exportReducer` as a 3rd parameter? We don't use that pattern elsewhere yet, but we'll probably have other cases like this where we need to send more data into a reducer. Then we can handle `setExportScale` entirely in the reducer